airflow/www/app.py

```diff
@@ -980,25 +984,27 @@ def action(self):
                     t.task_id
                     for t in task.get_flat_relatives(upstream=True)]
             TI = models.TaskInstance
+            dates = utils.date_range(start_date, end_date)
```

Comment:
I'm not sure what MAX_PERIODS should be, maybe we should limit what comes out of the cartesian product instead...